### ISSUE TITLE
Stream to twitch

### ISSUE THREAD

**Issue by louis030195** [PrismarineJS/prismarine-viewer#103]
Who needs human streamers when you can have bots ?
1. Stream to twitch a bot (Twitch API intregation)
2. Make him do some random stuff (outside of the scope of this repo)
3. Make him talk on the stream through text to speech based on its actions (outside of the scope of this repo)

**Comment by wvffle** [comment]
is it really required for `viewer` to be able to stream to twitch? It sounds like a use case and not a feature.

**Comment by rom1504** [comment]
yeah maybe not in the core lib but it could be an interesting example indeed!
https://github.com/PrismarineJS/prismarine-viewer/blob/master/examples/core/headless.js + some twitch lib

**Comment by rom1504** [comment]
seems there's currently no npm package to stream to twitch, however here's an example https://github.com/pion/example-webrtc-applications/tree/master/twitch

**Comment by Karang** [comment]
There is this feature in ffmpeg, so starting from the headless example this should be a 1 line change

**Comment by rom1504** [comment]
Ah yeah indeed, that's convenient.
https://wiki.archlinux.org/index.php/Streaming_to_twitch.tv#FFmpeg has
information on this

**Comment by joggerjoel** [comment]
Has this been solved? I’m interested in allowing the bot first person point of view. Even it can be played through a viewer then I can obs that source

**Comment by louis030195** [comment]
> Has this been solved? I’m interested in allowing the bot first person point of view. Even it can be played through a viewer then I can obs that source

@joggerjoel didn't try yet

**Comment by Karang** [comment]
This is done now, we have an example to show how to: https://github.com/PrismarineJS/prismarine-viewer/blob/master/examples/twitch.js

However this might need more testing to find the best parameters for here: https://github.com/PrismarineJS/prismarine-viewer/blob/master/lib/headless.js#L40

And maybe try to optimize the process to improve the framerate and video quality.
